### PR TITLE
 Structural changes and adding information.

### PR DESCRIPTION
**Pull request by Wisehh**
Changes made:

- Creation of a new section 'Tools and Editing':
  - The previous 'Tools and Editing' file is divided into multiple subsections (Eraser tool, Pen and Higlighter tools and LaTeX tool) that facilitate access to documentation on a specific tool. Additionally it lightens the content of the section since some tools have a very extensive documentation.
  - The documentation of 'Selection and Snapping tools' is added

- Creation of a new 'Plugins' section:
  - The previous Plugins file is divided into two new sections that facilitate punctual search, resulting in a subsection that documents the creation of plugins and another that talks about Notable Plugins (with an invitation to readers to enrich said subsection).

- Add information in 'Configuration':
  - Proposal for the structure of the section, where each section of the 'Preferences' menu corresponds to a subtitle with its respective documentation (perhaps if the information is very extensive later, it is advisable to create navigation subsections like the proposal made in the Tools section) .
  - Information about the autosave and the default name.
  - Information on the pressure of the Stylus.
  - Information about the configuration of buttons on the mouse and stylus.

- The pertinent changes are made in the 'mkdocs.yml' to reorder the structure.

- The 'toc.integrate' feature is added to 'mkdocs.yml' so that the table of contents is integrated into the navigation menu; giving more space to content and centralizing navigation in a single menu for greater ease.

Needs completing

- 'Autoloading Journals' in the first section of 'Preferences'
- ** tap to select ** configuration and how it works.
- Effect of snapping in the Spiline Drawing tool

### CODE REVIEW

File: docs/guide/tools/selecsnap.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+Rotation Snapping allows rotations of selected objects and strokes made with the tools mentioned above to be snapped to predetermined angles. These default angles correspond to multiples of 15 degrees.
+
+### Configuration
+
+Both tools can be adjusted their tolerance to snapping in the path: Edit> Preferences> Drawing Area> Snapping. Higher values correspond to greater intensity in the effect.
+
+The Snapping effect is disabled by default for shape recognition, but can be activated in the same section.
+
+Note that in the Grid Snapping description it is mentioned that the grid matches **by default** with the grid background. This caveat is made since in the path mentioned above, by means of the **Grid size** parameter it is possible to change the measurements of the grid by multiples of the original grid.
```
Comment (rolandlo) [inline docs/guide/tools/selecsnap.md:41]:
```suggestion
Note that in the Grid Snapping description it is mentioned that the grid matches **by default** the grid of the _Graph_ background. This grid consists of squares with side length 0.5 cm. By means of the **Grid size** parameter it is possible to change the side length of the grid squares as a multiple of the original length.
```

File: docs/guide/tools/selecsnap.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+In this way, for example, by setting 0.5 in the parameter, the internal grid becomes squares half the size of the grid background, so the objects will try to fit not only the lines of the squares, but also with its halves.
```
Comment (rolandlo) [inline docs/guide/tools/selecsnap.md:43]:
```suggestion
In this way, for example, by setting 0.5 in the parameter, the internal grid becomes squares half the size of the Graph paper grid background, so the objects will try to fit not only the lines of the squares, but also with its halves.
```

File: docs/guide/config.md [hunk 3]
```diff
@@ -1,3 +1,77 @@
 # Configuration
 
+# Preferences
+Xournal ++ has multiple options that can be configured and customized. Most of these features can be found in the path **Edit> Preferences **.
```
Comment (rolandlo) [inline docs/guide/config.md:4]:
```suggestion
Xournal ++ has multiple options that can be configured and customized. Most of these features can be found under **Edit > Preferences**.
```

File: docs/guide/config.md [hunk 4]
```diff
@@ -1,3 +1,77 @@
...
+
+TODO
+
+## Mouse
+
+In this section you can set the behavior of the **middle and right mouse buttons** to associate tools with it. Each tool allows sub settings for thickness, color, stroke type, etc. if they are available.
+
+## Stylus
+
+### Preassure Sensitivity
```
Comment (rolandlo) [inline docs/guide/config.md:30]:
```suggestion
### Pressure Sensitivity
```

**Review by rolandlo — CHANGES_REQUESTED** [review]
Thank you very much for your contribution. I have made a couple of improvement suggestions. Please look at them and add those you like to your commit (go to "Files Changed" and click on "Add to batch"). If you do not like the suggestion, try to see the point of my suggestion and try to make it better.

**Review by Wisehh — COMMENTED** [review]
Complete

**Comment by rolandlo** [comment]
Do you know how to squash your commits into one and rebase on current site branch? Here is [some info](https://www.internalpointers.com/post/squash-commits-into-one-git) on that.
Basically (having 2 commits to squash) you need to type
```
git rebase -i HEAD~2
git rebase upstream/site
git push --force
```
After the first command (interactive rebase) you will be put into editor mode where should mark the 2nd commit with `f` (like fixup). Save the file and exit the editor. Then type the other two git commands.

**Comment by Wisehh** [comment]
Now is the commit ok?

**Comment by rolandlo** [comment]
Thanks for squashing and rebasing. I merged it.

**Pull request merged**